packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -221,12 +221,12 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
         while (hierarchyParent && !findComponentByType(hierarchyParent, KnownComponentType.ModelRef)) {
           hierarchyParent = getSceneNodeByRef(hierarchyParent.parentRef);
         }
-        newHierarchyParentObject = getObject3DBySceneNodeRef(hierarchyParent?.ref);
+        newParentObject3D = getObject3DBySceneNodeRef(hierarchyParent?.ref);
       }
 
       let maintainedTransform: ITransform | null = null;
       if (originalObject3D) {
-        maintainedTransform = getFinalNodeTransform(originalObject, originalObject3D, newHierarchyParentObject);
+        maintainedTransform = getRelativeTransform(originalObject!, newParent!, getSceneNodeByRef);
```

Comment:
So this shouldn't be the relative transform (assuming relative equal nodes local transform)? This is instead the transform need to keep the object in the same place even though it has a new parent.   This function was working before for moving nodes and should still work for dynamic entities as written.  The issue is that https://github.com/awslabs/iot-app-kit/blob/main/packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx#L113 prevents the drag from happening if it's dynamic. But it looks like our loader isn't setting the reserved layer id on scene load so isDynamicNode is returning false which lets you drag anyways...